services/QuillLMS/client/app/bundles/Shared/components/shared/textEditor.tsx

```diff
@@ -100,11 +122,29 @@ class TextEditor extends React.Component <any, any> {
       return 'handled';
     }
     return 'not-handled';
-}
+  }
+
+  // this code was pasted from the tutorial here:
+  // https://medium.com/@siobhanpmahoney/building-a-rich-text-editor-with-react-and-draft-js-part-2-2-embedding-links-d71b57d187a7
+  handleAddLink = () => {
+    const { text } = this.state
+    const editorState = text;
+    const selection = editorState.getSelection();
+    const link = window.prompt('Paste the link -')
+    if (!link) {
+      this.handleTextChange(RichUtils.toggleLink(editorState, selection, null));
+      return 'handled';
+    }
+    const content = editorState.getCurrentContent();
+    const contentWithEntity = content.createEntity('LINK', 'MUTABLE', { url: link });
```

Comment:
Probably makes sense to constantize both 'LINK' and 'MUTABLE' so we can use the constants rather than the string literals.